src/api/include/migraphx/migraphx.hpp

```diff
@@ -786,28 +791,32 @@ struct module
         return instruction(ret_ins, own{});
     }
 
+    instruction add_allocation(const migraphx::shape& s)
+    {
+        migraphx_instruction_t ret_ins;
+        call(&migraphx_module_add_allocation, &ret_ins, mm.get(), s.get_handle_ptr());
+        return instruction(ret_ins, own{});
+    }
+
     migraphx_module_t get_handle_ptr() const { return mm.get(); }
 
     private:
     std::shared_ptr<migraphx_module> mm;
 };
 
-struct context
+struct context : MIGRAPHX_HANDLE_BASE(context)
 {
-    context(migraphx_context* p, borrow) : ctx(std::shared_ptr<migraphx_context*>(), p) {}
+    context() { this->make_handle(&migraphx_context_create); }
```

Comment:
Whats the reason for adding a default constructor?

Reply:
Removed, it is not needed.